### ISSUE TITLE
vectorization.h mismatch on MSVC

### ISSUE THREAD

**Issue by tjhei** [dealii/dealii#4976]
I am getting 
```
include\deal.II/base/vectorization.h(43): fatal error C1189: #error:  "Mismatch in vectorization capabilities: SSE2 was detected during configuration of deal.II and switched on, but it is apparently not available for the file you are trying to compile at the moment. Check compilation flags controlling the instruction set, such as -march=native."
```
not only on intel (https://github.com/dealii/dealii/pull/4440) but also on MSVC, see https://cdash.kyomu.43-1.org/viewBuildError.php?buildid=10837

Ideas?

**Comment by kronbichler** [comment]
The error message tells us that the MSVC compiler does not define the `__SSE2__` flag. Does somebody know what kind of flags the MSVC compiler does use to distinguish the vectorization? We could fix this by adding `#ifdef __SSE2__` to the tests in `cmake/checks/check_01_cpu_features.cmake` (and similarly for `AVX` and `AVX512`).

**Comment by tjhei** [comment]
only 
```
__AVX__
__AVX2__
```
according to https://msdn.microsoft.com/en-us/library/b0084kay.aspx but only if you specify ``/arch``, which we don't?

**Comment by tjhei** [comment]
(default for /arch is SSE2)

**Comment by kronbichler** [comment]
@tjhei I've added an improved check for the vectorization. For MSVC and SSE2 (but not AVX) we could also go the other route and remove that particular line from `vectorization.h`. What do you think?

**Comment by tjhei** [comment]
So this will disable all vectorization with MSVC for now, correct? We could also add ``!defined(DEAL_II_MSVC)`` just for SSE/SSE2, or not?

**Comment by tjhei** [comment]
I am assuming that vectorization works (and even AVX would work if we turn it on with ``/arch``)...

**Comment by kronbichler** [comment]
Yes, it will, because the check for `DEAL_II_COMPILER_VECTORIZATION_LEVEL` succeeds if that part is enabled. As an alternative, we might disable the check for `__SSE2__` and only rely on the facility in cmake through `DEAL_II_COMPILER_VECTORIZATION_LEVEL`. So let's try that in #4978.

**Comment by tjhei** [comment]
This looks okay to merge as is. Note that MSVC will define ``__AVX2__`` but not ``__AVX512F__``.

**Comment by kronbichler** [comment]
For `__AVX512F__` you need a KNL or an Intel Skylake Server CPU, which are both still quite rare.
